src/codec.rs

```diff
@@ -223,6 +223,13 @@ pub trait EncodeAppend {
 	fn append(self_encoded: Vec<u8>, to_append: &[Self::Item]) -> Result<Vec<u8>, Error>;
 }
 
+/// Trait that allows the length of a collection to be read, without having
+/// to read and decode the entire elements.
+pub trait EncodeLength {
+	/// Return the number of elements in `self_encoded`.
+	fn len(self_encoded: Vec<u8>) -> Result<u32, Error>;
```

Comment:
Please be a nice rust programmer and just take a slice :D

Reply:
Done but I more or less mimicked how `EncodeAppend` is treating the input. What's the rationale?

Reply:
`EncodeAppend` is another story. We try to reuse the given vector, as we append stuff to it directly. In this case, we just read a value. So it makes no sense to take the `Vec`.